polaris/polaris/sep10auth/views.py

```diff
@@ -89,20 +90,114 @@ def _transaction_from_post_request(request):
     return _get_transaction_json(request.body)
 
 
-def _validate_envelope_xdr(envelope_xdr):
+def _validate_challenge_xdr(envelope_xdr):
     """
     Validate the provided TransactionEnvelope XDR (base64 string). Return the
     appropriate error if it fails, else the empty string.
+
+    verify_challenge_transaction() does not verify that the transaction's
+    signers are valid and have the total weight greater or equal to the
+    source account's medium threshold value, so this function does that as
+    well.
     """
     try:
         verify_challenge_transaction(
```

Comment:
Yeah we need to add a couple test cases where 
```
Account A has signing weight
A: 0
B: 1
C: 1
and Threshold: 2
```
Case 1: Requests a token by signing with B and C succeeds, 
Case 2: Requests a token by signing with A fails
Case 3: Requests a token by signing with B fails
Case 4: Requests a token by signing twice with B fails (good catch on that leigh!)

Any other edge cases we should test for?

Reply:
If you'd like to identify more test cases, these are all the test cases I've added to the Go SDK. You can probably flesh these out in the actual Python SDK change though, because there's a lot.

https://github.com/stellar/go/pull/2071/files#diff-74afdec610280c6cd524e507134888daR1168

This also be a good time to work on updating the [transfer server validator](https://github.com/stellar/transfer-server-validator) so that its SEP-10 test suite tests all these cases. What do you think?

Reply:
I see, so the `verify_challenge_transaction` code needs to remove the call:
```
if not __verify_te_signed_by(transaction_envelope, manage_data_op.source)

Reply:
>I think `verify_challenge_transaction` should assume the client's source account has signed the transaction _unless_ `account_signers` and `threshold` are passed.

👍